aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -63,19 +69,40 @@ module AptosFramework::Token {
         amount: u64,
     }
 
+    /// create collection event with creator address and collection name
+    struct CreateCollectionEvent has drop, store {
+        creator: address,
+        collection_name: ASCII::String,
+    }
+
+    /// token creation event id of token created
+    struct CreateTokenEvent has drop, store {
+        id: TokenId,
+    }
+
+    /// token creation event id of token created
+    struct CreateNonFungibleTokenEvent has drop, store {
+        id: TokenId,
+    }
+
```

Comment:
Maybe just use a boolean to represent the max, but I don't think that this is required to be differentiated. Funny thing is most folks don't differentiate between the two. I think our code should just call them Tokens. We could try to qualify them, but not worth the hassle imo.

Reply:
I am not sure if other contract differentiate these two. But from end user view, they want to know if they are buying a NFT or fungible token.  I think @hariria mentioned this would be easier for wallet development. @hariria let me know if we want a separate event or having a max in the event is good enough?

Reply:
@davidiw @areshand we could also have it emit two events, one for creating a token, and one for creating an NFT. I think it would make indexing easier.

Reply:
I'm not sure how one would use the difference of the two. There might be nuances at a UX later well above the chain. Maybe let's start with one event and evaluate later? I suspect it's easier to add features and specialize over time than taking away something in use.